core/trino-main/src/main/java/io/trino/execution/SqlTaskExecution.java

```diff
@@ -569,9 +602,16 @@ public DriverSplitRunner createDriverRunner(@Nullable ScheduledSplit partitioned
             return new DriverSplitRunner(this, driverContext, partitionedSplit);
         }
 
+        @Nullable
```

Comment:
Maybe `Optional`. Also I wonder if it is necessary to avoid creating a driver. Maybe we can simply create one and let it go the normal flow.

Reply:
The only usage site is in `DriverSplitRunner` where `null` is cleaner to handle than `Optional`, so I'd prefer to keep this `@Nullable`.

Sure, we could create the driver anyway- but there's still the possibility that `DriverFactory#noMoreDrivers()` is called concurrently so we could fail to create the driver. Actually, I'll go ahead and update the logic to avoid the `taskStateMachine.getState().isTerminatingOrDone()` check but handle `DriverFactory#createDriver(DriverContext)` failing since that's the core real root of what we need to handle here by returning null.